test/java/seedu/addressbook/commands/AddCommandTest.java

```diff
@@ -0,0 +1,192 @@
+		for (String number : invalidNumbers) {
+			ConstructingInvalidAddCmdThrows(Name.EXAMPLE, number, false, Email.EXAMPLE, false, Address.EXAMPLE, false,
+					tags);
+		}
+	}
+
+	@Test
+	public void addCommand_invalidEmail_throws() {
+		Set<String> tags = new HashSet<String>();
+		final String[] invalidEmails = { "", " ", "def.com", "@", "@def", "@def.com", "abc@" };
+		for (String email : invalidEmails) {
+			ConstructingInvalidAddCmdThrows(Name.EXAMPLE, Phone.EXAMPLE, false, email, false, Address.EXAMPLE, false,
+					tags);
+		}
+	}
+
+	@Test
+	public void addCommand_invalidAddress_throws() {
+		Set<String> tags = new HashSet<String>();
+		final String[] invalidAddresses = { "", " " };
+		for (String address : invalidAddresses) {
+			ConstructingInvalidAddCmdThrows(Name.EXAMPLE, Phone.EXAMPLE, false, Email.EXAMPLE, false, address, false,
+					tags);
+		}
+	}
+
+	@Test
+	public void addCommand_invalidTags_throws() {
+		final String[][] invalidTags = { { "" }, { " " }, { "'" }, { "validTag", "" }, { "", " " } };
+		for (String[] tags : invalidTags) {
+			Set<String> tagsToAdd = new HashSet<String>();
+			for (String tag : tags) {
+				tagsToAdd.add(tag);
+			}
+			ConstructingInvalidAddCmdThrows(Name.EXAMPLE, Phone.EXAMPLE, false, Email.EXAMPLE, false, Address.EXAMPLE,
+					false, tagsToAdd);
+		}
+	}
+
+	/**
+	 * Asserts that attempting to construct an add command with the supplied
+	 * invalid data throws an IllegalValueException
+	 */
+	private void ConstructingInvalidAddCmdThrows(String name, String phone, boolean isPhonePrivate, String email,
+			boolean isEmailPrivate, String address, boolean isAddressPrivate, Set<String> tags) {
+		try {
+			new AddCommand(name, phone, isPhonePrivate, email, isEmailPrivate, address,
+					isAddressPrivate, tags);
+			final String errorFormatter = new String(
+					"An add command was successfully constructed with invalid input: %s %s %s %s %s %s %s %s");
+			String error = String.format(errorFormatter, name, phone, isPhonePrivate, email, isEmailPrivate, address,
+					isAddressPrivate, tags);
+			fail(error);
+		} 
+		catch (IllegalValueException e) {
+		}
+	}
+
+	@Test
+	public void addCommand_validData_correctlyConstructed() {
+		Set<String> tags = new HashSet<String>();
+		try {
+			AddCommand cmd = new AddCommand(Name.EXAMPLE, Phone.EXAMPLE, true, Email.EXAMPLE, false, Address.EXAMPLE,
+					true, tags);
+			ReadOnlyPerson p = cmd.getPerson();
+			
+			assertEquals(p.getName().fullName, Name.EXAMPLE);
+			assertEquals(p.getPhone().value, Phone.EXAMPLE);
+			assertTrue(p.getPhone().isPrivate());
+			assertEquals(p.getEmail().value, Email.EXAMPLE);
+			assertFalse(p.getEmail().isPrivate());
+			assertEquals(p.getAddress().value, Address.EXAMPLE);
+			assertTrue(p.getAddress().isPrivate());
+			boolean isTagListEmpty = !p.getTags().iterator().hasNext();
+			assertTrue(isTagListEmpty);
+		} 
+		catch (IllegalValueException e) {
+			fail("Adding command with valid data failed.");
+		}
```

Comment:
I think the try/catch can be omitted. Just let the exception go through (i.e. add a throws clause to the method signature). If the exception happens, it will result in a test failure anyway.